### PR TITLE
Expose the option --append-dns-names

### PR DESCRIPTION
**Pull request by lei-tang**
Expose the option --append-dns-names, which sets the CLI parameter append-dns-names of Citadel: 
- If true, Citadel will generate webhook certificates.
- If false, Citadel will not generate webhook certificates.
If there is another component generate webhook certificates, this parameter should be set as false.

Please provide a description for what this PR is for: #8736.

Design document: https://docs.google.com/document/d/1VHyBre8943USOx_YEVtA18KfEoGTmLT1iBHESO5bzcA

The matching PR on istio/installer repo is: https://github.com/istio/installer/pull/312.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: install/kubernetes/helm/istio/charts/security/values.yaml [hunk 1]
```diff
@@ -46,3 +46,11 @@ enableNamespacesByDefault: true
...
 podAntiAffinityLabelSelector: []
 podAntiAffinityTermLabelSelector: []
+
+# This parameter sets the CLI parameter append-dns-names of
+# Citadel, which:
+# - If true, Citadel will generate webhook certificates.
+# - If true, Citadel will not generate webhook certificates.
+# If there is another component generate webhook certificates,
+# this parameter should be set as false.
+appendDnsNames: true
```
Comment (howardjohn) [inline install/kubernetes/helm/istio/charts/security/values.yaml]: Why is the option named `appendDnsNames` when this option is about webhook config? I don't think it needs to directly line up with the CLI flag. This is user facing, while the cli flag isn't really

Comment (lei-tang) [reply install/kubernetes/helm/istio/charts/security/values.yaml]: Per the comment, the name of the option changed to enableGenerateWebhookCerts.

**Comment by costinm** [comment]
If another component like certmanager generates DNS certificates - I would rather change the install of galley/injector to customize the name of the cert it loads ( add extra volume, etc).

Galley will still need to mount an Istio cert (spiffee ) for the MCP port, so if citadel is not generating
the DNS cert it'll need to load 2 ports.

**Comment by costinm** [comment]
And if we do specify different ports - it doesn't matter that citadel is also adding the extra name.

**Comment by lei-tang** [comment]
Correct me if I am wrong, @myidpt: the certificate of Galley MCP server is moving to the Galley sidecar and is of SPIFFE identity format. 

The option here controls whether Citadel generates webhook certificates, which will not be used by Galley MCP server. The default value of enableGenerateWebhookCerts is true, which is the same as the current value of --append-dns-names option of Citadel. When Istio Webhook Controller is deployed, a user can set this option as false to disable Citadel generating webhook certificates, which simplifies/reduces the work of Citadel -> higher reliability.

File: install/kubernetes/helm/istio/charts/security/values.yaml [hunk 2]
```diff
@@ -46,3 +46,11 @@ enableNamespacesByDefault: true
 # "security" and value "S1".
 podAntiAffinityLabelSelector: []
 podAntiAffinityTermLabelSelector: []
+
+# This parameter sets the CLI parameter append-dns-names of
+# Citadel, which:
+# - If true, Citadel will generate webhook certificates.
+# - If true, Citadel will not generate webhook certificates.
```
Comment (morvencao) [inline install/kubernetes/helm/istio/charts/security/values.yaml]: should be `- If false, Citadel will not generate webhook certificates.` in the comment.

**Comment by rlenglet** [comment]
I met offline with @costinm and we agreed on the following plan of action:

1. We need a way to disable webhook configuration in Citadel and Injection Webhook, and such flags are the right way to implement feature flags in the `istio/istio` installer.
1. The flags can be added into `istio/istio` installer. **So we will submit this PR.**
1. The flags will not be added into the new `istio/installer`, since webhook configuration will be disabled by default in Citadel and Injection Webhook.
1. We need to test the logic and combinations of flags, especially upgrade from 1.3. **This is tracked in https://github.com/istio/istio/issues/16860.**

**Review by rlenglet — APPROVED** [review]

**Comment by rlenglet** [comment]
@istio/wg-environments-maintainers could you please review this PR?

**Comment by costinm** [comment]
And most important: istioctl and operator should add option to enable/disable validation and autoinjection, with a setting for all namespaces. This would add the webhook, after making sure that
Galley and Injector are ready and have certificate - and add the public key of citadel ( for current mode ) or apiserver or other CA, depending on user choice. I assume the simplest for the user is to just check what Galley and Injector are using - i.e. probe it and get the public key from the workload.

Re. which component will handle DNS certs for injector/galley - I don't think we have consensus on exclusive support for apiserver or running another component besides Citadel for cert provisioning. In an environment without citadel it might be ok to have a small component that only handles DNS certs using apiserver - but if Citadel is running I expect this component to also be linked in citadel, operational and install complexity are not worth it.  In any case - with citadel we have something that works and is stable.

**Comment by costinm** [comment]
I'm not involved in istio/istio installer review - so approval left for other people more familiar with the legacy installer. AFAIK we are still planning to deprecate it for 1.4 and use Operator/new installer - but in case we'll still ship old installer we must make sure upgrade/downgrade requirements are met, and anything dealing with autoinject config is extremely sensitive ( high risk )

**Comment by howardjohn** [comment]
still needed?

**Comment by lei-tang** [comment]
@howardjohn close for now; will reopen if it is needed.

**Pull request closed without merging**